src/main/java/edu/harvard/iq/dataverse/authorization/AuthenticationServiceBean.java

```diff
@@ -511,6 +533,69 @@ public AuthenticatedUser lookupUser( String apiToken ) {
         return tkn.getAuthenticatedUser();
     }
     
+    /*
+    getDeleteUserErrorMessages( AuthenticatedUser au )
+    method which checks for reasons that a user may not be deleted
+    -has created dvObjects
+    -has roles
+    -has guestbook records
+    -etc.
+    An empty string is returned if the user is 'deletable'
+    */
+    
+    public String getDeleteUserErrorMessages(AuthenticatedUser au) {
+        String retVal = "";
+        if (!dvObjSvc.findByAuthenticatedUserId(au).isEmpty()) {
+            retVal += BundleUtil.getStringFromBundle("admin.api.deleteUser.failure.dvobjects");
+        }
+
+        if (!roleAssigneeSvc.getAssignmentsFor(au.getIdentifier()).isEmpty()) {
+            if (!retVal.isEmpty()) {
+                retVal += "; ";
+            }
+            retVal += BundleUtil.getStringFromBundle("admin.api.deleteUser.failure.roleAssignments");
+        }
+
+        if (!gbRespSvc.findByAuthenticatedUserId(au).isEmpty()) {
+            if (!retVal.isEmpty()) {
+                retVal += "; ";
+            }
+            retVal += BundleUtil.getStringFromBundle("admin.api.deleteUser.failure.gbResps");
+        }
+
+        if (!datasetVersionService.getDatasetVersionUsersByAuthenticatedUser(au).isEmpty()) {
+            if (!retVal.isEmpty()) {
+                retVal += "; ";
+            }
+            retVal += BundleUtil.getStringFromBundle("admin.api.deleteUser.failure.versionUser");
+        }
+
+        if (!explicitGroupService.findGroups(au).isEmpty()) {
```

Comment:
For delete of user, I think it's be fine to also just remove them from groups at the time.

Reply:
done